rapid7_insightidr/komand_rapid7_insightidr/actions/close_investigations_in_bulk/action.py

```diff
@@ -0,0 +1,74 @@
+import komand
+from .schema import CloseInvestigationsInBulkInput, CloseInvestigationsInBulkOutput, Input, Output, Component
+from komand.exceptions import PluginException
+# Custom imports below
+from komand_rapid7_insightidr.util.endpoints import Investigations
+from komand_rapid7_insightidr.util.resource_helper import ResourceHelper
+import json
+from datetime import datetime
+from datetime import timedelta
+
+
+class CloseInvestigationsInBulk(komand.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name='close_investigations_in_bulk',
+            description=Component.DESCRIPTION,
+            input=CloseInvestigationsInBulkInput(),
+            output=CloseInvestigationsInBulkOutput())
+
+    def run(self, params={}):
+        request = ResourceHelper(self.connection.session, self.logger)
+        endpoint = Investigations.close_investigations_in_bulk(self.connection.url)
+
+        source = self._get_with_default(params, Input.SOURCE, "MANUAL")
+        max_investigations_to_close = self._get_with_default(params, Input.MAX_INVESTIGATIONS_TO_CLOSE, None)
+        alert_type = self._get_with_default(params, Input.ALERT_TYPE, None)
+
+        timestamp_from = params.get(Input.DATETIME_FROM)
+        if not timestamp_from:
+            timestamp_from = (datetime.now() - timedelta(days=7)).replace(microsecond=0).isoformat()
+            timestamp_from = f"{timestamp_from}Z"
+
+        timestamp_to = params.get(Input.DATETIME_TO)
+        if not timestamp_to:
+            timestamp_to = datetime.now().replace(microsecond=0).isoformat()
+            timestamp_to = f"{timestamp_to}Z"
+
+        self.logger.info(f"{timestamp_from}")
+
+        response = request.resource_request(
+            endpoint,
+            'post',
+            payload={
+                "alert_type": alert_type,
+                "from": timestamp_from,
+                "max_investigations_to_close": max_investigations_to_close,
+                "source": source,
+                "to": timestamp_to,
+            }
+        )
+
+        try:
+            result = json.loads(response.get("resource", '{}'))
```

Comment:
Is this right? Usually, I see this as:
 
response.json().get("resource")

Reply:
API return object like this:
{
    'response': '{"ids": ["xxx"], "num_close": 1}'
}

So API return dict where in one key there is string which is json.
So I'm getting here "response" from dict: `response.get("resource", '{}')`, with default value when string will be empty.
Next I'm parse json string to dict. Then I can return value with dict. 

In other actions there is similar use case.

Reply:
OK cool, thanks for the clarification.